datastream-server/src/main/java/com/linkedin/datastream/server/DatastreamTaskImpl.java

```diff
@@ -436,8 +436,25 @@ public int hashCode() {
   @Override
   public String toString() {
     // toString() is mainly for logging purpose, feel free to modify the content/format
-    return String.format("%s(%s), partitionsV2=%s, partitions=%s, dependencies=%s", getDatastreamTaskName(), _connectorType,
-        String.join(",", _partitionsV2), LogUtils.logNumberArrayInRange(_partitions), _dependencies);
+    // If _partitions is populated, it is replicated into _partitionsV2 for assignment. However,
+    // if _partitions is not populated, we can assume that only _partitionsV2 is populated.
```

Comment:
I think more context is necessary here for when _partitions vs. _partitionsV2 is used. Partition managed uses _partitionsV2 and puts in the full name of the Topic-Partition, due to which the log cannot be reduced in this way.

I'd also question whether we should avoid populating _partitionsV2 altogether for any tasks that are not Partition managed. This can be explored as part of another PR though (maybe open a ticket for this)

Reply:
Added comments. Also created ticket to investigate.